kedro/io/data_catalog.py

```diff
@@ -141,6 +173,7 @@ def __init__(
         data_sets: dict[str, AbstractDataSet] = None,
```

Comment:
Maybe we need to align the spelling here in 0.19. It's quite evident here:

- `data_sets`
- `dataset_patterns`